service/storage_service.cc

```diff
@@ -1737,10 +1742,8 @@ storage_service::prepare_replacement_info(std::unordered_set<gms::inet_address>
 
     auto dc_rack = get_dc_rack_for(replace_address);
 
-    // use the replacee's host Id as our own so we receive hints, etc
-    auto host_id = _gossiper.get_host_id(replace_address);
-    co_await _sys_ks.local().set_local_host_id(host_id).discard_result();
-    const_cast<db::config&>(_db.local().get_config()).host_id = host_id; // FIXME -- carry non-cost config on storage service itself
+    auto replace_host_id = _gossiper.get_host_id(replace_address);
+    slogger.info("Host {}/{} is replacing {}/{} using a different address", _db.local().get_config().host_id, get_broadcast_address(), replace_host_id, replace_address);
```

Comment:
If the fix is so simple - we can just remove these lines - can ew expedite queuing it outside everything else? There are a lot of dependencies on it in Raft and we would be able to significantly simplify the address map of Raft with it.

Reply:
@asias attempts that in https://github.com/scylladb/scylladb/pull/12078
by using the existing node's tokens as a way for it to realize it was replaced if it comes back from the dead.
I have reservations about that method, see https://github.com/scylladb/scylladb/pull/12078/files#r1035637006

Reply:
Regardless of the method, what is the connection between detecting a zombie node that's come back and not reusing the host id? Is reusing the host id today a way to detect this condition?